tools/rpkg/R/register.R

```diff
@@ -52,11 +52,12 @@ duckdb_register_arrow <- function(conn, name, arrow_scannable) {
   stopifnot(dbIsValid(conn))
 
     # create some R functions to pass to c-land
-    export_fun <- function(arrow_scannable, stream_ptr) {
-        record_batch_reader <- arrow::Scanner$create(arrow_scannable)$ToRecordBatchReader()
-        record_batch_reader$export_to_c(stream_ptr)
+    export_fun <- function(arrow_scannable, stream_ptr, projection=NULL, filter=TRUE) {
+        arrow::Scanner$create(arrow_scannable, projection, filter)$ToRecordBatchReader()$export_to_c(stream_ptr)
```

Comment:
FYI, if `arrow_scannable` is the result of previous dplyr queries on an Arrow object, `Scanner$create()` will [use the filter and projection stored in that object](https://github.com/apache/arrow/blob/master/r/R/dataset-scan.R#L82-L97) and will ignore your filter and projection args here